pyuvsim/tests/test_source.py

```diff
@@ -101,3 +106,49 @@ def test_pol_coherency_calc():
     nt.assert_true(np.allclose(coh_loc[1, 1, unpol_srcs_up], 0.5))
     nt.assert_true(np.allclose(coh_loc[1, 0, unpol_srcs_up], 0.0))
     nt.assert_true(np.allclose(coh_loc[0, 1, unpol_srcs_up], 0.0))
+
+
+def test_read_healpix_hdf5():
+    Nside = 32
+    Npix = hp.nside2npix(Nside)
+    vec = hp.ang2vec(np.pi / 2, np.pi * 3 / 4)
+    ipix_disc = hp.query_disc(nside=32, vec=vec, radius=np.radians(10))
+    m = np.arange(Npix)
+    m[ipix_disc] = m.max()
+
+    indices = np.arange(Npix)
+
+    frequencies = 100000000 * np.ones(len(indices))
+
+    hpmap, inds, freqs = pyuvsim.source.read_healpix_hdf5(os.path.join(SIM_DATA_PATH, 'test_file.hdf5'), 0)
+
+    nt.assert_true(np.allclose(hpmap, m))
+    nt.assert_true(np.allclose(inds, indices))
+    nt.assert_true(np.allclose(freqs, frequencies))
+
+
+def test_healpix_to_sky():
+    Nside = 32
+    Npix = hp.nside2npix(Nside)
+    vec = hp.ang2vec(np.pi / 2, np.pi * 3 / 4)
+    ipix_disc = hp.query_disc(nside=32, vec=vec, radius=np.radians(10))
+    m = np.arange(Npix)
+    m[ipix_disc] = m.max()
+
+    hpmap, inds, freqs = pyuvsim.source.read_healpix_hdf5(os.path.join(SIM_DATA_PATH, 'test_file.hdf5'), 0)
+    m = m / simutils.jy2Tsr(freqs, bm=hp.nside2pixarea(Nside), mK=False)
+    sky = pyuvsim.source.healpix_to_sky(hpmap, inds, freqs)
+
+    nt.assert_true(np.allclose(sky.stokes[0], m))
+
+
+def test_units_healpix_to_sky():
+    Nside = 32
+    beam_area = hp.pixelfunc.nside2pixarea(Nside) * units.sr
+    hpmap, inds, freqs = pyuvsim.source.read_healpix_hdf5(os.path.join(SIM_DATA_PATH, 'test_file.hdf5'), 0)
+    freqs = freqs * units.Hz
+    stokes = (hpmap * units.K).to(units.Jy, units.brightness_temperature(beam_area, freqs))
+    sky = pyuvsim.source.healpix_to_sky(hpmap, inds, freqs)
+    print(sky.stokes[0])
+    print(stokes.value)
+    nt.assert_true(np.allclose(sky.stokes[0], stokes.value))
```

Comment:
I'd like to see one more test that takes a healpix map with only one non-zero pixel and compares the visibility to an analytic calculation.